cpp/src/io/utilities/datasource.cpp

```diff
@@ -227,14 +228,27 @@ class memory_mapped_source : public file_source {
   }
 
  private:
+  [[nodiscard]] bool should_register_mmap_buffer()
+  {
+    if (_map_addr == nullptr) { return false; }
+
+    auto const policy = getenv_or("LIBCUDF_MMAP_REGISTER_ENABLED", std::string{"AUTO"});
+
+    if (policy == "ALWAYS") { return true; }
+    if (policy == "AUTO") { return pageableMemoryAccessUsesHostPageTables(); }
+    if (policy == "OFF") { return false; }
+
+    CUDF_FAIL("Invalid LIBCUDF_MMAP_REGISTER_POLICY value: " + policy);
+  }
+
   /**
    * @brief Page-locks (registers) the memory range of the mapped file.
    *
    * Fixes nvbugs/4215160
    */
   void register_mmap_buffer(size_t offset, size_t size)
   {
-    if (_map_addr == nullptr or not pageableMemoryAccessUsesHostPageTables()) { return; }
+    if (not should_register_mmap_buffer()) { return; }
```

Comment:
Probably we should throw here instead of silently return, to avoid misconfigured to register mmap?

Reply:
no need to throw, registering is optional. By default we only do it on GH systems.

Reply:
But what if the users actually want to register (so they call this function) but forgot to set the config value to register?

Reply:
I see what you mean. However, this is not public API. The only way to express that registration is required is to set the env var.